rsub.sh

```diff
@@ -12,62 +37,73 @@ usage() {
 	exit 1
 }
 
-trap '' HUP
-append=1
+init() {
+	if ! check_absolute_path "$target"; then
+		>&2 echo "rsub: $target is not an absolute path"
+		exit 1
+	fi
 
-while getopts Al:r: arg; do
-	case "$arg" in
-		A) append=0 ;;
-		r) line_regex="$OPTARG" ;;
-		l) line_text="$OPTARG" ;;
-		?) usage ;;
-	esac
-done
-shift $(($OPTIND - 1))
-[ $# -eq 1 ] || usage
+	test -f "$target" || {
+		>&2 echo "rsub: file not found: $target"
+		exit 3
+	}
 
-target=$1
+	source=$(mktemp rsub_XXXXXXXX)
 
-case $(dirname "$target") in
-	/*) ;;
-	*)
-		>&2 echo "rsub: $target is not an absolute path"
-		exit 1
-		;;
-esac
+	trap '' HUP
+}
 
-test -f "$target" || {
-	>&2 echo "rsub: file not found: $target"
-	exit 3
+parse_args() {
+	while getopts Al:r: arg; do
+		case "$arg" in
+			A) append=0 ;;
+			r) line_regex="$OPTARG" ;;
+			l) line_text="$OPTARG" ;;
+			?) usage ;;
+		esac
+	done
+	shift $(($OPTIND - 1))
+	[ $# -eq 1 ] || usage
```

Comment:
It looks like you used `test` everywhere except this line. Myabe it's worth to also replace it with `test`